app/lib/features/comments/widgets/comment.dart

```diff
@@ -1,28 +1,81 @@
+import 'package:acter/common/providers/room_providers.dart';
 import 'package:acter/common/widgets/render_html.dart';
+import 'package:acter_avatar/acter_avatar.dart';
 import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart';
+import 'package:dart_date/dart_date.dart';
 import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:acter/features/comments/widgets/skeletons/comment_item_skeleton_widget.dart';
 
-class CommentWidget extends StatelessWidget {
+class CommentWidget extends ConsumerWidget {
   final Comment comment;
-  const CommentWidget({super.key, required this.comment});
+  final CommentsManager manager;
+
+  const CommentWidget({
+    super.key,
+    required this.comment,
+    required this.manager,
+  });
 
   @override
-  Widget build(BuildContext context) {
+  Widget build(BuildContext context, WidgetRef ref) {
     final msgContent = comment.msgContent();
     final formatted = msgContent.formattedBody();
-
-    return Card(
-      child: Column(
-        crossAxisAlignment: CrossAxisAlignment.stretch,
-        children: [
-          ListTile(title: Text(comment.sender().toString())),
-          formatted != null
-              ? RenderHtml(
-                  text: formatted,
-                )
-              : Text(msgContent.body()),
-        ],
+    var commentTime =
+        DateTime.fromMillisecondsSinceEpoch(comment.originServerTs());
+    final time = commentTime.timeago();
+    final memberInfo = ref.watch(
+      roomMemberProvider(
+        (roomId: manager.roomIdStr(), userId: comment.sender().toString()),
       ),
     );
+
+    return memberInfo.when(
+      data: (data) {
+        final profileData = data.profile;
+        return Card(
+          child: Column(
+            crossAxisAlignment: CrossAxisAlignment.stretch,
+            children: [
+              ListTile(
+                leading: ActerAvatar(
+                  mode: DisplayMode.DM,
+                  avatarInfo: AvatarInfo(
+                    uniqueId: manager.roomIdStr(),
+                    displayName: profileData.displayName ?? manager.roomIdStr(),
+                    avatar: profileData.getAvatarImage(),
+                  ),
+                  size: 18,
+                ),
+                title: Text(
+                  profileData.displayName ?? manager.roomIdStr(),
```

Comment:
here, too. Probably safe to to just grep that once as a variable and just assign it ...
```suggestion
                  profileData.displayName ?? userId,
```